File: crux_core/typegen_extensions/swift/requests.swift
import Serde


public extension [Request] {
  static func bincodeDeserialize(input: [UInt8]) throws -> [Request] {
    let deserializer = BincodeDeserializer(input: input)
    try deserializer.increase_container_depth()
    let length = try deserializer.deserialize_len()

    var requests: [Request] = []
    for _ in 0 ..< length {
      while deserializer.get_buffer_offset() < input.count {
        let req = try Request.deserialize(deserializer: deserializer)
        requests.append(req)
      }
    }
    deserializer.decrease_container_depth()

    return requests
  }
}

File: crux_core/typegen_extensions/kotlin/Requests.kt
object Requests {

    @JvmStatic
    @Throws(com.novi.serde.DeserializationError::class)
    fun bincodeDeserialize(input: ByteArray?): List<Request> {
        if (input == null) {
            throw com.novi.serde.DeserializationError("Cannot deserialize null array")
        }

        val deserializer: com.novi.serde.Deserializer = com.novi.bincode.BincodeDeserializer(input)
        deserializer.increase_container_depth()

        val length = deserializer.deserialize_len()

        val value = mutableListOf<Request>()

        for (i in 0 until length) {
            value.add(Request.deserialize(deserializer))
        }

        deserializer.decrease_container_depth()

        if (deserializer.get_buffer_offset() < input.size) {
            throw com.novi.serde.DeserializationError("Some input bytes were not read")
        }
        return value
    }
}
